packages/rollup-contracts/contracts/ExecutionManager.sol

```diff
@@ -573,24 +573,18 @@ contract ExecutionManager is FullStateManager {
      * @return True if this succeeded, false otherwise.
      */
     function createNewContract(address _newOvmContractAddress, bytes memory _ovmInitcode) internal returns (bool){
-        // Safety check the initcode -- unless the overrideSafetyChecker flag is set to true
-        if (!overrideSafetyChecker && !safetyChecker.isBytecodeSafe(_ovmInitcode)) {
-            // Contract init code is not safe.
-            return false;
-        }
         // Switch the context to be the new contract
         (address oldMsgSender, address oldActiveContract) = switchActiveContract(_newOvmContractAddress);
         // Deploy the _ovmInitcode as a code contract -- Note the init script will run in the newly set context
-        address codeContractAddress = deployCodeContract(_ovmInitcode);
-        // Get the runtime bytecode
-        bytes memory codeContractBytecode = getCodeContractBytecode(codeContractAddress);
-        // Safety check the runtime bytecode -- unless the overrideSafetyChecker flag is set to true
-        if (!overrideSafetyChecker && !safetyChecker.isBytecodeSafe(codeContractBytecode)) {
-            // Contract runtime bytecode is not safe.
+        address codeContractAddress = stateManager.deployContract(_newOvmContractAddress, _ovmInitcode);
```

Comment:
Hmmmm I'm still weirded out by doing purity checking in the SM as it feels much more "execution-ey".  However we have recorded the need to revisit this for pre-deployment of code contracts so probably fine for now.

Reply:
Tbh I have the opposite feeling wrt purity checking lol but will of course see what happens as the partial state manager is implemented